samples/bot-sso-adaptivecard/csharp/BotSsoAdaptivecard/Bots/DialogBot.cs

```diff
@@ -0,0 +1,275 @@
+
+            // TODO : Use token or state to perform operation on behalf of user
+
+            string[] filepath = { ".", "Resources", fileName };
+
+            var adaptiveCardJson = File.ReadAllText(Path.Combine(filepath));
+            AdaptiveCardTemplate template = new AdaptiveCardTemplate(adaptiveCardJson);
+            var authResultData = isTokenPresent ? "SSO success" : isStatePresent ? "OAuth success" : "SSO/OAuth failed";
+            if (isBasicRefresh)
+            {
+                authResultData = "Refresh done";
+            }
+            var payloadData = new
+            {
+                authResult = authResultData,
+            };
+
+            var cardJsonstring = template.Expand(payloadData);
+
+            var adaptiveCardResponse = new AdaptiveCardInvokeResponse()
+            {
+                StatusCode = 200,
+                Type = AdaptiveCard.ContentType,
+                Value = JsonConvert.DeserializeObject(cardJsonstring)
+            };
+            return CreateInvokeResponse(adaptiveCardResponse);
+        }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="turnContext"></param>
+        /// <param name="cancellationToken"></param>
+        /// <returns></returns>
+        private async Task<InvokeResponse> initiateSSOAsync(ITurnContext<IInvokeActivity> turnContext, CancellationToken cancellationToken)
+        {
+            var signInLink = await GetSignInLinkAsync(turnContext, cancellationToken).ConfigureAwait(false);
+            var oAuthCard = new OAuthCard
+            {
+                Text = "Signin Text",
+                ConnectionName = _connectionName,
+                TokenExchangeResource = new TokenExchangeResource
+                {
+                    Id = Guid.NewGuid().ToString()
+                },
+                Buttons = new List<CardAction>
+                    {
+                        new CardAction
+                        {
+                            Type = ActionTypes.Signin,
+                            Value = signInLink,
+                            Title = "Please sign in",
+                        },
+                    }
+            };
+
+
+            var loginReqResponse = JObject.FromObject(new
+            {
+                statusCode = 401,
+                type = "application/vnd.microsoft.activity.loginRequest",
+                value = oAuthCard
+            });
+
+            return CreateInvokeResponse(loginReqResponse);
+        }
+
+        /// <summary>
+        /// Get Adaptive Card
+        /// </summary>
+        /// <param name="filepath"></param>
+        /// <param name="signInLink"></param>
+        /// <param name="name"></param>
+        /// <param name="userMRI"></param>
+        /// <returns></returns>
+        private Attachment GetFirstOptionsAdaptiveCard(string[] filepath, string signInLink, string name = null, string userMRI = null)
```

Comment:
nit: rename to GetAdaptiveCardWithAuthOptions

Reply:
I have updated your comment. Could you please review and update